src/prototype_room_creepbuilder.js

```diff
@@ -51,8 +50,10 @@ Room.prototype.spawnCreateCreep = function(role, heal, level, squad, routing, ba
     if (returnCode != name) {
       continue;
     }
-    let userName = Memory.username || _.find(Game.spawns, 'owner').owner;
     if (config.stats.enabled) {
+      let userName = Memory.username || _.find(Game.spawns, 'owner').owner;
+      Memory.stats = Memory.stats || {};
+      Memory.stats[userName].roles = Memory.stats[userName].roles || {};
       let roleStat = Memory.stats[userName].roles[role];
       let previousAmount = roleStat ? roleStat : 0;
       Memory.stats[userName].roles[role] = previousAmount + 1;
```

Comment:
`brain.stats.add(['roles', role], previousAmount + 1);`
could work so i supose and we do no more need name define and previous init i think.
there is the decrease one in brain_memory l.55. you could change it the same i believe.